### ISSUE TITLE
Could not locate bindings file

### ISSUE THREAD

**Issue by stanleynguyen** [cdimascio/express-openapi-validator#159]
I've just upgraded to `2.18.0` and faced with this issue

```
    Could not locate the bindings file. Tried:
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/build/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/build/Debug/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/build/Release/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/out/Debug/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/Debug/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/out/Release/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/Release/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/build/default/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/compiled/10.16.0/darwin/x64/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/addon-build/release/install-root/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/addon-build/debug/install-root/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/addon-build/default/install-root/deasync.node
     → /Users/stanleynguyen/Documents/Projects/template-node/{{cookiecutter.service_name}}/node_modules/deasync/lib/binding/node-v64-darwin-x64/deasync.node

      at bindings (node_modules/bindings/bindings.js:126:9)
      at Object.<anonymous> (node_modules/deasync/index.js:30:31)
      at Object.<anonymous> (node_modules/express-openapi-validator/dist/framework/json.ref.schema.js:4:19)
```

Does anyone know what might be the potential cause?

**Comment by cdimascio** [comment]
Can you try running `npm rebuild`

**Comment by stanleynguyen** [comment]
@cdimascio same thing happen. My node version is `v10.16.0` and npm version is `6.13.1` if these are helpful

**Comment by cdimascio** [comment]
thanks @stanleynguyen. what os are you running?
fyi: i'm looking into removing the dependency on deasync.

**Comment by cdimascio** [comment]
One more thing to try https://github.com/nodejs/node-gyp/wiki/Updating-npm's-bundled-node-gyp
Let me know how it goes.

Note that I am looking to remove the dependency on deasync

**Comment by stanleynguyen** [comment]
Same thing still happened after I updated npm's node-gyp. I'm using macOS.
On another note, we are facing the same issue on our CI running with base image of Docker node 10 image.

**Comment by cdimascio** [comment]
@stanleynguyen which docker image are you using?

Note that the project uses both the Node 10 and Node 12 docker images as provided by Travis to install dependencies and execute tests. In both cases, all dependencies are successfully install and all tests pass. see [here](https://travis-ci.org/cdimascio/express-openapi-validator)

**Comment by cdimascio** [comment]
FYI: I did a quick test using the following docker image `node:10-alpine` and had no issue.

**Comment by stanleynguyen** [comment]
That's strange 🤔 We are using the same image as well
The new version without deasync sounds awesome! If this's too much of a hassle for you to check, we can always wait for the next version 👍
Btw there's an extra piece of information, I'm using some internal APIs of your package. For the previous version (without deasync error), it's something like this
```js
this.openApiValidator = new ExpressValidator(opts);
const ctx = new OpenApiContext({ apiDoc: opts.apiSpec });
this.manualValidator = createResponseAjv(ctx.apiDoc);
```

With the recent update, I've been trying this to as there were some changes happened
```js
this.openApiValidator = new ExpressValidator(opts);
const spec = new OpenApiSpecLoader({ apiDoc: opts.apiSpec }).load();
const ctx = new OpenApiContext(spec, undefined);
this.manualValidator = createResponseAjv(ctx.apiDoc);
```

**Comment by cdimascio** [comment]
@stanleynguyen you can try the new version now. I've released it as alpha in order to get feedback. It is feature complete.

See the [alpha readme](https://github.com/cdimascio/express-openapi-validator/tree/async_1) for details. Please provide feedback on the new api, etc

**Comment by stanleynguyen** [comment]
Hey @cdimascio , I have tried 3.0.0 and it's working perfectly. However, imho, the async way of constructing and installing the validator into the app feels a bit weird and less flexible for me (like an unnecessary evil). However, I'm fully behind deasync removal from the package.

**Comment by cdimascio** [comment]
Yes. the blocking parse certainly makes usage more straightforward. unfortunately to parse json ref's, asynchronicity is necessary. the synchronous install is still part of v3 and can be used with `installSync`. that being said, `installSync` requires `deasync` and as we've seen, deasync has been at the root of many recent issues.

**Comment by cdimascio** [comment]
fixed in v3.0.0
